TowerForge/TowerForge/LevelManager/Components/AiComponent.swift

```diff
@@ -8,9 +8,14 @@
 import Foundation
 import UIKit
 
+protocol AiSpawnDelegate: AnyObject {
+    func spawnAiUnit<T: TFEntity & PlayerSpawnable>(ofType type: T.Type, position: CGPoint)
+}
+
```

Comment:
Hmm maybe it is better to add a delegate for snapping into grid position so that anyone that needs to spawn anything is able to normalise their position? Possibly add the normalisation delegate in the attemptSpawn method that I have added in HomeSystem?